src/app/main/component/user/components/profile/profile-cards/profile-cards.component.ts

```diff
@@ -23,8 +26,27 @@ export class ProfileCardsComponent implements OnInit, OnDestroy {
 
   ngOnInit() {
     this.languageSubscription = this.localStorageService.languageBehaviourSubject.subscribe(() => {
+      this.loadFactsFromLocalStorage();
       this.getFactOfTheDay();
+      this.checkAndUpdateHabitFact();
     });
+
+    this.dailyUpdateSubscription = timer(0, 24 * 60 * 60 * 1000).subscribe(() => {
+      this.checkAndUpdateHabitFact();
+    });
+  }
+
+  loadFactsFromLocalStorage() {
```

Comment:
I presume that this logic should be in the service

Reply:
Yes, you are right, I planned to do that and redo the tests